cmake/version.cmake

```diff
@@ -16,23 +16,26 @@ set(PROJECT_VERSION_TWEAK ${CMAKE_MATCH_1})
 string(REGEX MATCH "EXTRAVERSION = ([a-z0-9]*)" _ ${ver})
 set(PROJECT_VERSION_EXTRA ${CMAKE_MATCH_1})
 
+# Temporary convenience variable
+set(PROJECT_VERSION_WITHOUT_TWEAK ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH})
 
-message("Zephyr version: ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
-
-set(PROJECT_VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH})
 if(PROJECT_VERSION_TWEAK)
-  set(PROJECT_VERSION "${PROJECT_VERSION}.${PROJECT_VERSION_TWEAK}")
+  set(PROJECT_VERSION ${PROJECT_VERSION_WITHOUT_TWEAK}.${PROJECT_VERSION_TWEAK})
+else()
+  set(PROJECT_VERSION ${PROJECT_VERSION_WITHOUT_TWEAK})
 endif()
 
+message("Zephyr version: ${PROJECT_VERSION_WITHOUT_TWEAK}")
```

Comment:
why not set this to PROJECT_VERSION you just determined in above line?

Reply:
Because I wanted to make this a pure refactoring without making any changes.

Also, I had no way of knowing why the original author omitted the tweak.

Fixed.